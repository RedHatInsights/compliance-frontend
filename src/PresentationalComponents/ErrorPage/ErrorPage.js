import React from 'react';
import propTypes from 'prop-types';
import InvalidObject from '@redhat-cloud-services/frontend-components/InvalidObject';
import NotAuthorized from '@redhat-cloud-services/frontend-components/NotAuthorized';
import { ErrorState } from '@redhat-cloud-services/frontend-components/ErrorState';

const ErrorPage = ({ error }) => {
  if (error.networkError && error.networkError.statusCode === 401) {
    window.insights.chrome.auth.logout(true);
    return false;
  }

  if (error.networkError && error.networkError.statusCode === 403) {
    return <NotAuthorized serviceName="Compliance" />;
  }

  if (error.networkError && error.networkError.statusCode === 404) {
    return <InvalidObject />;
  }

  return <ErrorState />;
};

ErrorPage.propTypes = {
  error: propTypes.object,
};

export default ErrorPage;
