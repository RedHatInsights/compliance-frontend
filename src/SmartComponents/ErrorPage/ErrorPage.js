import { Component } from 'react';
import propTypes from 'prop-types';

class ErrorPage extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { error } = this.props;
        if (error.networkError.statusCode === 401) {
            window.insights.chrome.auth.logout();
        }

        return 'Oops! There was an error loading Compliance data. If you need to contact ' +
            'Red Hat Support about this, this is the exact error: ' +
            error.networkError.statusCode + ' ' + error;
    }
}

ErrorPage.propTypes = {
    error: propTypes.object
};

export default ErrorPage;
