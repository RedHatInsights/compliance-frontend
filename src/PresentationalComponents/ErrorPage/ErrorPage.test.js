import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import useChrome from '@redhat-cloud-services/frontend-components/useChrome';

jest.mock('@redhat-cloud-services/frontend-components/useChrome', () => ({
  __esModule: true,
  default: jest.fn(),
}));

import ErrorPage from './ErrorPage';

describe('ErrorPage', () => {
  it('expect to render an invalid object', () => {
    const error = {
      networkError: { statusCode: 404 },
      error: 'Not found',
    };
    render(<ErrorPage error={error} />);

    expect(screen.getByText('We lost that page')).toBeInTheDocument();
  });

  it('expect to render a 403 error', () => {
    const error = {
      networkError: { statusCode: 403 },
      error: 'Not authorized',
    };
    render(<ErrorPage error={error} />);

    expect(
      screen.getByText('You do not have access to Compliance')
    ).toBeInTheDocument();
  });

  it('expect to render a 401 error when logged out', () => {
    const logout = jest.fn(() => 'Logout');
    useChrome.mockImplementation(() => ({ auth: { logout } }));
    const error = {
      networkError: { statusCode: 401 },
      error: 'Test Error loading',
    };
    render(<ErrorPage error={error} />);

    expect(logout).toHaveBeenCalled();
  });
});
