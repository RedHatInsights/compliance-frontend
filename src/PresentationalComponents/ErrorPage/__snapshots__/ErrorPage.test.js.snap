// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage expect to render a 401 error when logged out 1`] = `""`;

exports[`ErrorPage expect to render a 403 error 1`] = `
<NotAuthorized
  serviceName="Compliance"
/>
`;

exports[`ErrorPage expect to render an invalid object 1`] = `<InvalidObject />`;

exports[`ErrorPage expect to render without error 1`] = `<ErrorState />`;
